update meshkernel and hydrolib-core versions in toml

veenstrajelmer opened the pull request:
## Issue addressed
Fixes #207

## Explanation
Hydrolib-core was updated to support meshkernel 6.*. This PR also enforces these newer versions in hydromt_delft3dfm.

## Checklist
- [x] Updated tests or added new tests >> not relevant
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed >> fixed docs workflow
- [x] Updated changelog.rst if needed >> not relevant

shartgring commented:
Thanks Jelmer! Maybe the tests are failing due to a similar issue as in hydromt_wflow (https://github.com/Deltares/hydromt_wflow/issues/328)? Will keep an eye on it if we find anything over there

shartgring commented:
I changed it in a commit, and now the tests are passing. However doc building fails, not sure why though

veenstrajelmer commented:
Fixed! Updated to py 3.12, avoided the warning about the defaults channel and set `sphinx<8.2`. This new sphinx version was released right before you kicked off the docs action: https://github.com/sphinx-doc/sphinx/releases/tag/v8.2.0. This upper limit has to be removed again, so I have created https://github.com/Deltares/hydromt_delft3dfm/issues/209 for this.

shartgring reviewed (approved):
LGTM
